src/Dependencies/PooledObjects/ArrayBuilder.cs

```diff
@@ -318,6 +322,27 @@ public ImmutableArray<U> ToDowncastedImmutable<U>()
             return tmp.ToImmutableAndFree();
         }
 
+#if COMPILERCORE
```

Comment:
>COMPILERCORE [](http://example.com/codeflow?start=4&length=12)

I think making this an extension method would be better.

Reply:
i'm trying to both exactly match the methods thsee are based on (in this case, ToImmutableAndFree), and also have them be in the same location so the behaviors/impls can be compared/understood.

Reply:
These sources are included in code in other repos, as well as any 3rd party repo that references the source package. Having COMPILERCORE show up there is rather undesirable.

Reply:
i mean, i'd be ok with OneOrMore being in that source package :)   we have it referenced in other layers anyways :)

Reply:
I don't quite get why not use extension method here. It's rather trivial impl.

Reply:
my goal was to keep this where the existing helpers were that it was aping.  But i can move it :)